src/twisted/web/test/test_static.py

```diff
@@ -155,6 +155,34 @@ def test_emptyChild(self):
         self.assertEqual(child.path, base.path)
 
 
+    def test_emptyChildUnicodeParent(self):
+        """
+        The C{u''} child of a L{File} which corresponds to a directory
+        whose path is text is a L{DirectoryLister} that renders to a
+        binary listing.
+
+        @see: U{https://twistedmatrix.com/trac/ticket/9438}
+        """
+        textBase = FilePath(self.mktemp()).asTextMode()
+        textBase.makedirs()
+        textBase.child(u"text-file").open('w').close()
+        textFile = static.File(textBase.path)
+
+        request = DummyRequest([b''])
+        child = resource.getChildForRequest(textFile, request)
+        self.assertIsInstance(child, static.DirectoryLister)
+
+        nativePath = compat.nativeString(textBase.path)
+        self.assertEqual(child.path, nativePath)
+
+        response = child.render(request)
+        self.assertIsInstance(response, bytes)
+    if sys.getfilesystemencoding().lower() not in ('utf-8', 'mcbs'):
+        test_emptyChildUnicodeParent.skip = (
+            "Cannot write unicode filenames with file system encoding of"
```

Comment:
ITYM "non-ASCII"?  Given that the `u''` child is the one we care about, I'm pretty sure that will survive most boneheaded filesystem encodings, and this skip could be removed?

Reply:
This is a safeguard I carried over from [another test](https://github.com/twisted/twisted/blob/7ae65314dcc50979e10f70d6ad4dfd1f67d06786/src/twisted/web/test/test_static.py#L312-L339), because who knows what [sys.getfilesystemencoding](https://docs.python.org/2/library/sys.html#sys.getfilesystemencoding), which `FilePath` uses, is going to return.  But as you point out, it should be impossible that `SynchronousTestCase.mkstemp().decode(sys.getfilesystemencoding()) + u''` could return an unencodable path, so this skip is unnecessary.